torch_geometric/nn/position_encoding.py

```diff
@@ -0,0 +1,50 @@
+from typing import Union
+
+import torch
+import torch.nn.functional as F
+
+
+class PositionEncoding1D(torch.nn.Module):
+    r"""The 1D positioning encoding from `"Attention Is All You Need"
+    <https://arxiv.org/pdf/1706.03762.pdf>`_ paper.
+    """
+    def __init__(self, num_pos: int, out_channels: int,
+                 max_wave_len: Union[float, int] = 10000., mode: str = "sin"):
+        super().__init__()
+
+        self.mode = mode
+        self.num_pos = num_pos
+        self.out_channles = out_channels
+
+        if self.mode == "sin":
+            if out_channels % 2 != 0:
+                raise ValueError(
+                    "Cannot use sinusoidal positional encoding with "
+                    f"odd dim. Got `out_channles`: {out_channels}).")
+
+            position_encoding = torch.zeros(num_pos, out_channels)
+
+            position = torch.arange(0, num_pos, dtype=torch.float).unsqueeze(1)
```

Comment:
I wonder whether we can support floating-point positions, which would let us support positions of different granularity. For this, we would likely need to drop "learn" support (not sure how tragic this would be), and perform sinus/cosinus computation on the fly. WDYT?

Reply:
I am no expert TBH but I felt that the range of positions can still be given via `num_pos` (or better via `max_pos`), right? I don't think we can normalize by max on a per-batch basis.

Reply:
Add a simple `granularity` support. The users can change the granularity if the floating-point positions are scaled or if they want to capture more stable changes in floating-point positions.